src/kernel.cpp

```diff
@@ -373,7 +373,18 @@ bool ComputeNextStakeModifier(const CBlockIndex* pindexPrev, uint64_t& nStakeMod
 // Get stake modifier checksum
 unsigned int GetStakeModifierChecksum(const CBlockIndex* pindex)
 {
-    assert (pindex->pprev || pindex->GetBlockHash() == (!fTestNet ? hashGenesisBlock : hashGenesisBlockTestNet));
+    if (pindex->pprev == nullptr) 
+    {
+        if(pindex->GetBlockHash() != (!fTestNet ? hashGenesisBlock : hashGenesisBlockTestNet)) 
```

Comment:
Saw the discussion is Slack. Since we're tweaking this area, we might as well get rid of the unnecessary and confusing hash check from the old code:

```suggestion
    if (pindex->pprev == nullptr && pindex != pindexGenesisBlock) 
    {
```

Reply:
@RoboticMind That patch probably broke the build. I didn't undo the nested block so that it wouldn't interfere with Jim's review comment. You will probably probably need to remove the indentation and closing brace.

Reply:
Should have fixed it now, did a force push to update your commit.

Reply:
Are you guys sure we want to get rid of the hash check? it is being done elsewhere for sure.

Reply:
Someone just used the hash to determine whether the cursor reached the beginning of the chain (this comes from Peercoin). A node formally verifies the block index hashes elsewhere during init. In fact, `GetStakeModifierChecksum()` doesn't actually do anything for Gridcoin (it's broken and we have no stake modifier checkpoints)&mdash;I left it alone during my last pass through the kernel code because that assertion has proven to be useful as a canary for a corrupt chain. I cleaned this stuff up in another branch where I'm working on block index improvements, but this PR is still applicable.

Reply:
Roger! Then the pointer comparison is more than adequate here.